--- a/arpwatch.c	2013-09-11 09:09:58.129397693 +0200
+++ b/arpwatch.c	2013-09-11 09:26:43.600415691 +0200
@@ -174,6 +174,7 @@ main(int argc, char **argv)
 	struct bpf_program code;
 	char errbuf[PCAP_ERRBUF_SIZE];
 	char* serveruser = NULL;
+	int promisc = 1;
 
 	if (argv[0] == NULL)
 		prog = "arpwatch";
@@ -191,7 +192,7 @@ main(int argc, char **argv)
 	interface = NULL;
 	rfilename = NULL;
 	pd = NULL;
-	while ((op = getopt(argc, argv, "df:i:n:Nr:u:e:s:")) != EOF)
+	while ((op = getopt(argc, argv, "df:i:n:Nr:u:e:s:p")) != EOF)
 		switch (op) {
 
 		case 'd':
@@ -253,6 +254,10 @@ main(int argc, char **argv)
 			}
 			break;
 
+		case 'p':
+			--promisc;
+			break;
+
 		default:
 			usage();
 		}
@@ -320,7 +325,7 @@ main(int argc, char **argv)
 		snaplen = max(sizeof(struct ether_header),
 		    sizeof(struct fddi_header)) + sizeof(struct ether_arp);
 		timeout = 1000;
-		pd = pcap_open_live(interface, snaplen, 1, timeout, errbuf);
+		pd = pcap_open_live(interface, snaplen, promisc, timeout, errbuf);
 		if (pd == NULL) {
 			syslog(LOG_ERR, "pcap open %s: %s", interface, errbuf);
 			exit(1);
--- a/arpwatch.8	2013-09-11 09:34:43.042486395 +0200
+++ b/arpwatch.8	2013-09-11 09:09:58.124397690 +0200
@@ -27,7 +27,7 @@ arpwatch - keep track of ethernet/ip add
 .na
 .B arpwatch
 [
-.B -dN
+.B -dNp
 ] [
 .B -f
 .I datafile
@@ -70,6 +70,10 @@ background and emailing the reports. Ins
 .IR stderr .
 .LP
 The
+.B -p
+flag disables promiscous mode.
+.LP
+The
 .B -f
 flag is used to set the ethernet/ip address database filename.
 The default is
